proposals/records-wip.md

```diff
@@ -99,3 +99,4 @@ of the left-hand side.
 The evaluation of a `with` expression is equivalent to calling the `Clone` method exactly once,
 and then setting the backing field of each record property in the argument list to its corresponding
```

Comment:
>record property [](start = 43, length = 15)

I don't know what a "record property" is. Also, the return type of the `Clone` method might not be a type that has the named property as a member.

Reply:
Defined a record property above. The record property that is compiler-generated if one of equivalent signature does not exist.

Reply:
There is no requirement that it be a record type. Moreover, I don't know what it means for a type to be a record type from the point of view outside the type itself.

---
In reply to: [406952410](https://github.com/dotnet/csharplang/pull/3353#discussion_r406952410) [](ancestors = 406952410)

Reply:
If it's not a record type, there are no generated record members. It's just a consequence of the precondition.

Reply:
Are you saying that the "record"ness of a type and the "generated record property"ness of a property, are visible to the language externally and across compilation boundaries?  If so, I would expect to see that in the implementation as well.

---
In reply to: [406953742](https://github.com/dotnet/csharplang/pull/3353#discussion_r406953742) [](ancestors = 406953742)

Reply:
That would seem to be true! However, the implementation is a prototype, and only implements a subset of the spec 😄

Reply:
I thought you were only specifying what you're implementing.

---
In reply to: [406956349](https://github.com/dotnet/csharplang/pull/3353#discussion_r406956349) [](ancestors = 406956349)

Reply:
Alternatively, I could phrase this as "the property is `init`-only, but I don't have a full specification there either, so it seems like either way we'll end up with some specification gap.

This phrasing makes sense... it's just broader than will eventually be implemented.